collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py

```diff
@@ -767,10 +890,20 @@ def generate_pods_dict(array):
                 'mediator': pods[pod]['mediator'],
                 'mediator_version': pods[pod]['mediator_version'],
             }
-        pods_fp = array.list_pods(failover_preference=True)
-        for pod in range(0, len(pods_fp)):
-            acpod = pods_fp[pod]['name']
-            pods_info[acpod]['failover_preference'] = pods_fp[pod]['failover_preference']
+            if pods_info[acpod]['arrays'][0]['frozen_at']:
```

Comment:
Suggest to change pods_info[acpod]['arrays'][0]['frozen_at'] to pods_info[acpod]['arrays'][0].get('frozen_at') for backwards compatibility with purity 5.x.

Reply:
Moved to a better location where version check is applied